Mathlib/Topology/IsClosedRestrict.lean

```diff
@@ -0,0 +1,140 @@
+/-
+Copyright (c) 2025 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne, Peter Pfaffelhuber
+-/
+import Mathlib.Topology.Homeomorph
+
+/-! # Restriction of a closed compact set in a product space to a set of coordinates
+
+We show that the image of a compact closed set `s` in a product `Π i : ι, α i` by
+the restriction to a subset of coordinates `S : Set ι` is a closed set.
+
+The idea of the proof is to use `isClosedMap_snd_of_compactSpace`, which is the fact that if
+`X` is a compact topological space, then `Prod.snd : X × Y → Y` is a closed map.
+
+We remark that `s` belongs to the set `Sᶜ.restrict  ⁻¹' (Sᶜ.restrict '' s)`, and we build
+an homeomorphism `Sᶜ.restrict ⁻¹' (Sᶜ.restrict '' s) ≃ₜ Sᶜ.restrict '' s × Π i : S, α i`.
+`Sᶜ.restrict '' s` is a compact space since `s` is compact, and the lemma applies,
+with `X = Sᶜ.restrict '' s` and `Y = Π i : S, α i`.
+
+-/
+
+open Set
+
+variable {ι : Type*} {α : ι → Type*} {s : Set (Π i, α i)} {i : ι} {S : Set ι}
+
+namespace Homeomorph
```

Comment:
I'm not sure it's the right namespace: most of the lemmas below don't deal with homeomorphisms, it's just the final construction which is a homeo.

Reply:
I put everything except for the final def in the `Topology` namespace